src/components/Footer.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+import { deleteCompletedTodos } from '../api/todos';
+import { useTodosContext } from './useTodosContext';
+
+type Props = {
+  query: Status;
+  setQuery: (query: Status) => void;
+};
+
+export const Footer: React.FC<Props> = ({ query, setQuery }) => {
+  const { list, setList, setLoadingTodosIds, handleError } = useTodosContext();
+  const completedIds = list
+    .filter(item => item.completed === true)
+    .map(item => item.id);
+  const notCompletedQuantity = list.filter(
+    (todo: Todo) => !todo.completed,
+  ).length;
+
+  const handleClearCompleted = () => {
+    setLoadingTodosIds(completedIds);
```

Comment:
Add new loading todos to the previous one, use functional `setState` in case you have other todos loading at the same time.
And do the same in all other places where you set or remove loading todo ids